### ISSUE TITLE
Internal Server Error, TemplateSyntaxError: non-default argument follows default argument

### ISSUE THREAD

**Issue by thias** [voxpupuli/puppetboard#346]
This is with f016820, latest commit on master as of right now (continuation of #345, same environment as that report).

For the "Overview" initial page, I get the top menu, but the content of the page is an "Internal Server Error" message, with the following logged : 
```
[2017-02-01 09:48:39,613] ERROR in app: Exception on / [GET]
Traceback (most recent call last):
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
response = self.full_dispatch_request()
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
rv = self.dispatch_request()
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/home/puppetboard/.local/lib/python2.7/site-packages/puppetboard/app.py", line 256, in index
current_env=env
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/templating.py", line 134, in render_template
context, ctx.app)
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/templating.py", line 116, in _render
rv = template.render(context)
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
return self.environment.handle_exception(exc_info, True)
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
reraise(exc_type, exc_value, tb)
File "/home/puppetboard/.local/lib/python2.7/site-packages/puppetboard/templates/_macros.html", line 93, in template
{% macro datatable_init(table_html_id, ajax_url, default_length, length_selector, extra_options=None, columns) -%}
TemplateSyntaxError: non-default argument follows default argument
[pid: 19447|app: 0|req: 16/16] 2001:920:5846:9:e6f5:d26b:8012:2664 () {46 vars in 1064 bytes} [Wed Feb  1 09:48:39 2017] GET / => generated 3311 bytes in 251 msecs (HTTP/1.1 500) 2 headers in 100 bytes (1 switches on core 0)
```
For the "Nodes" tab, I get a blank page, with a different error logged, but about the same macro file+line : 
```
Traceback (most recent call last):
File "/home/puppetboard/.local/lib/python2.7/site-packages/werkzeug/wsgi.py", line 703, in __next__
return self._next()
File "/home/puppetboard/.local/lib/python2.7/site-packages/werkzeug/wrappers.py", line 81, in _iter_encoded
for item in iterable:
File "/home/puppetboard/.local/lib/python2.7/site-packages/flask/helpers.py", line 130, in generator
for item in gen:
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1271, in __next__
return self._next()
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1248, in _buffered_generator
c = next(self._gen)
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1045, in generate
yield self.environment.handle_exception(exc_info, True)
File "/home/puppetboard/.local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
reraise(exc_type, exc_value, tb)
File "/home/puppetboard/.local/lib/python2.7/site-packages/puppetboard/templates/_macros.html", line 93, in template
{% macro datatable_init(table_html_id, ajax_url, default_length, length_selector, extra_options=None, columns) -%}
jinja2.exceptions.TemplateSyntaxError: non-default argument follows default argument
[pid: 19447|app: 0|req: 17/17] 2001:920:5846:9:e6f5:d26b:8012:2664 () {46 vars in 1074 bytes} [Wed Feb  1 09:49:54 2017] GET /nodes => generated 0 bytes in 150 msecs (HTTP/1.1 200) 1 headers in 59 bytes (0 switches on core 0)
```

**Comment by redref** [comment]
Really strange behavior. My jinja2 version is much more conciliant (2.7.2).

There is already a fix in #341

**Comment by thias** [comment]
OK, thanks! Indeed, see http://jinja.pocoo.org/docs/dev/changelog/#version-2-8 with : 

_"Added a check for default arguments followed by non-default arguments. This change makes {% macro m(x, y=1, z) %}...{% endmacro %} a syntax error. The previous behavior for this code was broken anyway (resulting in the default value being applied to y)"_

**Comment by thias** [comment]
This has been fixed with the changes pushed from #341.
